---
title: "P1 GWAS Catalog - Systemic lupus Erythematosus"
author: "Ana Cecilia González Álvarez"
output: 
  flexdashboard::flex_dashboard:
    theme: journal
---


```{r setup, include=FALSE}
library(ggplot2)
library(flexdashboard)
library(plotly)
library(hrbrthemes)
library(DT) 


##Filtered by disease and empty chromosomes removed ~30,000
lupus_cat <- read.csv("/Users/gonzalac/OneDrive - KAUST/gwas_lupus_filtrado.csv")
lupus_cat ##1071
##Filtramos for p values
lupus_cat <- lupus_cat[lupus_cat$P.VALUE < 1e-6,]
lupus_cat  ##731

##Ordenamos X axis
lupus_cat$CHR_ID <- factor(lupus_cat$CHR_ID, levels = c("1","2","3","4","5","6","7","8","9",
                  "10","11","12","13","14","15","16","17","18","19","20","21","22","X","Y"))
  

```

Column {data-width=600}
-------------------------------------
    
### Manhattan Plot
    
```{r}
p <- (ggplot(data=lupus_cat, aes(x=CHR_ID, y=PVALUE_MLOG, color=CHR_ID, 
        SNPS=SNPS, REGION=REGION, P.VALUE = P.VALUE, DATE = DATE, PUBMEDID = PUBMEDID)) + 
        geom_point(size = 3, alpha = 0.55) + labs(x = "Genomic Position (number denotes chromosome)", 
        y = "-log10 p-value") + 
        theme_classic() + theme(legend.position="none"))
ggplotly(p, tooltip = c("SNPS", "REGION", "P.VALUE","DATE", "PUBMEDID")) 

```

Column {data-width=400}
-------------------------------------
   
### Table

```{r}
datatable(lupus_cat, rownames = FALSE, options = list(pageLength =5))
```

### Bar Plot
```{r}
b <- ggplot(data = lupus_cat, aes(CONTEXT)) + geom_bar(color = "indianred2", fill = "indianred2")
b + theme_classic() + 
          coord_flip() + labs(x="Variant type", y = "Count") +  
          scale_x_discrete(limit =c("synonymous_variant", "regulatory_region_variant", 
                                    "non_coding_transcript_exon_variant","missense_variant", "intron_variant", "intergenic_variant"), 
                           labels = c("Synonymous","Reg region","Exon", "Missense","Intron", "Intergenic"))

```


